In [1]:
from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score

In [13]:
# Load-Split-Instantiate-fit-predict-score
X, y = datasets.load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.8, random_state=42) 
clf = SVC()
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)
accuracy_score(y_test, y_pred)

0.9333333333333333

A very strange phenomena is that whenever we use random_state 42 it almost gives us a perfect model with an accuracy of 1.0 (applicable to varied set of models, lr,svm,knn etc). Even for 20:80 split of training:test it gives 0.93 accuracy. Is it because the random_state=42 was used when these models were tested and optimized. Is this something specific to iris dataset?